internal/process/provisioning/create_kyma_name_step.go

```diff
@@ -0,0 +1,45 @@
+package provisioning
+
+import (
+	"fmt"
+	"github.com/kyma-project/kyma-environment-broker/internal/process/steps"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+)
+
+type CreateKymaNameStep struct {
+	operationManager *process.OperationManager
+}
+
+func NewCreateKymaNameStep(os storage.Operations) *CreateKymaNameStep {
+	return &CreateKymaNameStep{
+		operationManager: process.NewOperationManager(os),
+	}
+}
+
+func (s *CreateKymaNameStep) Name() string {
+	return "Create_Kyma_Name"
+}
+
+func (s *CreateKymaNameStep) Run(operation internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if time.Since(operation.UpdatedAt) > CreateRuntimeTimeout {
+		log.Infof("operation has reached the time limit: updated operation time: %s", operation.UpdatedAt)
+		return s.operationManager.OperationFailed(operation, fmt.Sprintf("operation has reached the time limit: %s", CreateRuntimeTimeout), nil, log)
+	}
+
+	if operation.RuntimeID == "" {
+		return s.operationManager.OperationFailed(operation, fmt.Sprint("RuntimeID not set, cannot create Kyma name"), nil, log)
+	}
+
+	// could be simplified but this provides single source of truth for Kyma name
+	operation.KymaResourceName = ""
+	operation.KymaResourceName = steps.KymaName(operation)
```

Comment:
Is that duplicated assignment required?

Reply:
I do not understand why line 40 is required. I guess it is just to force KymaResourceName to be empty so that `steps.KymaName` for sure sets it to `RuntimeId`. Is that correct? It seems confusing because we need to get into details of invoked method when analysing what the empty assignment is all about.

Reply:
I have the same feeling. It seems awkward. I need to reconsider this with @piotrmiskiewicz if this is what we wanted and if we have to pay such a price for this `single point of truth`.
I would prefer to change steps.KymaName() semantics and avoid `emptying` the operation.KymaResourceName.
But this function is already used exactly in this way.

Reply:
I tried to eat a cake and have a cake. Please take a look.